src/qt/walletcontroller.cpp

```diff
@@ -137,11 +135,24 @@ WalletModel* WalletController::getOrCreateWallet(std::unique_ptr<interfaces::Wal
     connect(wallet_model, &WalletModel::coinsSent, this, &WalletController::coinsSent);
 
     // Notify walletAdded signal on the GUI thread.
-    Q_EMIT walletAdded(wallet_model);
-
+    if (QThread::currentThread() == thread()) {
+        addWallet(wallet_model);
+    } else {
+        // Handler callback runs in a different thread so fix wallet model thread affinity.
+        wallet_model->moveToThread(thread());
+        bool invoked = QMetaObject::invokeMethod(this, "addWallet", Qt::QueuedConnection, Q_ARG(WalletModel*, wallet_model));
```

Comment:
I was going to suggest to use `Qt::BlockingQueuedConnection` instead, so that the caller gets the wallet model with the parent already set.

Reply:
- works without blocking. 
- dont like blocking in general, if something goes wrong u thread is stuck, but need to think about.

Reply:
i think - it stucks on windows if u do Blocking. Need to check more, but not today.